pymatgen/io/cif.py

```diff
@@ -533,6 +544,36 @@ def _sanitize_data(self, data):
             for final_key, interim_key in changes_to_make.items():
                 data.data[final_key] = data.data[interim_key]
 
+        """
+        This part of the code deals with cases where _atom_site_type_symbol
+        contains symbols with all capital letters (e.g. CA instead of Ca).
+        Common in the COD database.
+        """
+
+        if "_atom_site_type_symbol" in data.data.keys():
+            # _atom_site_type_symbol capitalization with str.title().
+            # Some exception will remain with both capital letters.
+            no_capitalization_symbols = ["OH", "NH"]
+            new_atom_site_type_symbols = []
+            for sym in data.data["_atom_site_type_symbol"]:
+                if any(sym.startswith(c) for c in no_capitalization_symbols):
+                    new_atom_site_type_symbols.append(sym)
+                else:
+                    new_atom_site_type_symbols.append(sym.title())
+
+            data.data["_atom_site_type_symbol"] = new_atom_site_type_symbols
+        elif '_atom_site_label' in data.data.keys():
+            r = re.compile("|".join(upper_case_elements))
+
+            for i, sym in enumerate(data.data['_atom_site_label']):
+                match = r.match(sym)
+                if match:
+                    new_sym = sym.replace(match.group(), match.group().title())
+                    msg = "label {} converted to {}".format(sym, new_sym)
+                    warnings.warn(msg)
+                    self.errors.append(msg)
+                    data.data['_atom_site_label'][i] = new_sym
+
```

Comment:
I really wouldn't add a whole section of if statements to deal with this. I would rather deal with such issues in the section that parses the element symbols and allow all caps symbols there, with a warning if needed.